google/cloud/opentelemetry/internal/time_series_test.cc

```diff
@@ -170,6 +170,74 @@ TEST(SumPointData, NonEmptyInterval) {
   EXPECT_THAT(ts.points(), ElementsAre(Interval(start, expected_end)));
 }
 
+TEST(LastValuePointData, Simple) {
+  auto const now = std::chrono::system_clock::now();
+
+  opentelemetry::sdk::metrics::MetricData md;
+  md.instrument_descriptor.unit_ = "unit";
+  md.instrument_descriptor.value_type_ =
+      opentelemetry::sdk::metrics::InstrumentValueType::kInt;
+  md.start_ts = now;
+  md.end_ts = now;
+
+  opentelemetry::sdk::metrics::LastValuePointData point;
+  point.value_ = 42L;
+
+  auto interval = [](std::chrono::system_clock::time_point end) {
+    return AllOf(
+        ResultOf(
+            "has_start_time",
+            [](google::monitoring::v3::Point const& p) {
+              return p.interval().has_start_time();
+            },
+            Eq(false)),
+        ResultOf(
+            "end_time",
+            [](google::monitoring::v3::Point const& p) {
+              return internal::ToChronoTimePoint(p.interval().end_time());
+            },
+            Eq(end)));
+  };
+
+  auto ts = ToTimeSeries(md, point);
+  EXPECT_EQ(ts.unit(), "unit");
+  EXPECT_EQ(ts.metric_kind(), google::api::MetricDescriptor::GAUGE);
+  EXPECT_THAT(ts.points(),
+              ElementsAre(AllOf(Int64TypedValue(42), interval(now))));
+}
+
+TEST(LastValuePointData, IntValueTypes) {
+  for (auto value_type : {
+           opentelemetry::sdk::metrics::InstrumentValueType::kInt,
+           opentelemetry::sdk::metrics::InstrumentValueType::kLong,
+       }) {
+    opentelemetry::sdk::metrics::MetricData md;
```

Comment:
Here and below, you may want to use `SCOPED_TRACE` to facilitate debugging.

Reply:
Your point is that GoogleTest will only give us a line number and nothing in the expectation says which `value_type` failed.

Added `SCOPED_TRACE()` for these tests and the tests from the previous PR. Don't tell the code archaeologists.